### DISCUSSION TITLE
Build GraphQL API at runtime

### DISCUSSION THREAD

**Discussion by ad-si** [morpheusgraphql/morpheus-graphql#641] [Q&A]
Does Morpheus support building a GraphQL API during runtime (e.g. based on user input)? 
I'm doubtful since it uses TemplateHaskell to generate the resolver types.

**Comment by nalchevanidze** [comment]
Hi @ad-si, with [app package](https://github.com/morpheusgraphql/morpheus-graphql/tree/master/morpheus-graphql-app) you can build dynamic apps. is it what you are looking for?

**Reply by ad-si** [reply]
No. I need to provide the GraphQL schema at runtime! The link you provided describes the schema as a TemplateHaskell string at compile time.

**Reply by nalchevanidze** [reply]
No, template Haskell is not necessary at all. We can do same also without it. Only difference will be that we will get type errors at runtime

**Reply by ad-si** [reply]
I'm probably looking at the wrong spot, but I couldn't find anything in the documentation. So, e.g. I have following runtime definition of the API (just to use the same example as your documentation). How do I go from here?

```hs
let
  deity =
    [ ("fullName", "text")
    , ("power", "maybe text")
    ]
  deityArgs =
    [ ("name", "text")
    , ("mythology", "maybe text")
    ]
in 
  ?
```

**Reply by nalchevanidze** [reply]
I will update documentation and post here

**Comment by nalchevanidze** [comment]
@ad-si checkout readme again  https://github.com/morpheusgraphql/morpheus-graphql/tree/master/morpheus-graphql-app

**Reply by ad-si** [reply]
Cool, thanks for your efforts!
I still need to provide the `schema.gql` at runtime, which I don't have / would have to create in Haskell. Does Morpheus provide a way to assemble the schema in Haskell?
